packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -45,12 +46,46 @@ const ConditionGroupBuilder = ({
       variant == 'tree'
         ? handleFocusOnCloseTree(evt)
         : handleFocusOnClose(evt, conditionIndex);
-      onChange({
-        ...group,
-        conditions: group.conditions.filter(
-          (condition) => conditionId !== condition.id
-        ),
-      });
+
+      if (group.conditions[1].conditions) {
+        //when we remove every plain conditions of a group without deleting  the subgroup, we need to restructure the group
+        //the inner group become outer group and same level subgroups become plain conditions
+
+        let newGroup = [],
+          allConditions = [];
+        group.conditions.map((eachItem) => {
```

Comment:
instead of using `map` you can use `reduce` similarly to the way you do below to simplify this
```
const allConditions = group.conditions.reduce((acc, item) => {
  if (item.conditions) {
    return acc.concat(item.conditions);
  }
  return acc;
}, []);
```

Reply:
modified as advised